### DISCUSSION TITLE
Configure Git with my name and email

### DISCUSSION THREAD

**Discussion by Taona93** [community/community#139663] [New to GitHub]
### Select Topic Area

Show & Tell

### Body

I am trying to commit a project to git and I am getting a fatal error: unable to auto-detect email address and username. I need help on setting up my username and email address on VS

**Comment by fruetalo182** [comment]
Hi @Taona93 
You just need to set your Github username and email in a terminal on your project. Something like this:
`cd Github/my-project`
`git config --global user.name "myusername"`
`git config --global user.email "my-email@me.com"`

**Comment by K-a-r-e-e-m** [comment]
Hi @Taona93,
You can clone the repository in the following ways:

1- Using Username and Email:
git clone https://github.com/your_username/your_repo.git
After cloning, you can configure your username and email with:
git config --global user.name "your_username"
git config --global user.email "[your_email@example.com](mailto:your_email@example.com)"

2- Using a GitHub Token: If you have a GitHub token, you can use it in the URL for cloning:
git clone https://[your_token@github.com](mailto:your_token@github.com)/your_username/your_repo.git

This allows you to push changes directly without encountering authentication errors.
